mm/regions.c

```diff
@@ -96,6 +101,29 @@ bool has_memory_range(paddr_t pa) {
     return mbi_get_memory_range(pa, NULL) == 0;
 }
 
+static bool ranges_overlap(addr_range_t *range_a, addr_range_t *range_b) {
+    return (range_a->start <= range_b->end) && (range_b->start <= range_a->end);
```

Comment:
probably should be exclusive range (`<=` -> `<`) ?

Reply:
Please also squash relevant update commits and fix the commit message (headline, description with max 80-100 char lines, proper signoff-by)

Reply:
In the future, we probably should add another flag `reserved` to the `frame_t` and handle it properly in the PMM. Then, maybe instead of skipping the frames, we could create them and reserve (consume) them immediately. Same should apply to all the memory regions reported as reserved by multiboot, ACPI and others.